core/src/main/java/org/elasticsearch/common/geo/GeoDistance.java

```diff
@@ -72,7 +72,20 @@ public static GeoDistance readFromStream(StreamInput in) throws IOException {
     public void writeTo(StreamOutput out) throws IOException {
         Version clientVersion = out.getVersion();
         int ord = this.ordinal();
-        if (clientVersion.before(Version.V_5_3_3)) {
+        if (clientVersion.onOrAfter(Version.V_5_3_0) && clientVersion.before(Version.V_5_3_3)) {
+            switch (ord) {
+                case 0:
+                    out.write(0);  // write PLANE ordinal
+                    return;
+                case 1:
+                    out.write(1);  // write bwc ARC ordinal
+                    return;
+                default:
+                    throw new IOException("Unknown GeoDistance ordinal [" + ord + "]");
+            }
+        }
+
+        if (clientVersion.before(Version.V_5_3_0)) {
```

Comment:
I think it would be easier to reason about it it was expressed as something like 
```
if (clientVersion.before(Version.V_5_3_0)) {
...
} else if (clientVersion.before(Version.V_5_3_3)) {
...
} else {
...
}
```

There is also no  `else` after this statement, which means we write ord twice... And if tests don't catch this, we might need to figure out how to write a better test that would.

Reply:
thanks, I was *this* close to refactoring it, but decided to be lazy. thanks for keeping me honest. I've updated